app/src/main/java/org/greenstand/android/TreeTracker/fragments/ConfigFragment.kt

```diff
@@ -0,0 +1,73 @@
+package org.greenstand.android.TreeTracker.fragments
+
+import android.os.Bundle
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import kotlinx.android.synthetic.main.activity_main.toolbarTitle
+import kotlinx.android.synthetic.main.fragment_config.convergenceDataSizeVal
+import kotlinx.android.synthetic.main.fragment_config.convergenceTimeoutVal
+import kotlinx.android.synthetic.main.fragment_config.latStdDevThresholdVal
+import kotlinx.android.synthetic.main.fragment_config.locationDataConfig
+import kotlinx.android.synthetic.main.fragment_config.lonStdDevThresholdVal
+import kotlinx.android.synthetic.main.fragment_config.minDisBtwnUpdatesVal
+import kotlinx.android.synthetic.main.fragment_config.minTimeBtwnUpdatesVal
+import kotlinx.android.synthetic.main.fragment_config.saveConfigButton
+import org.greenstand.android.TreeTracker.R
+import org.greenstand.android.TreeTracker.models.LocationDataConfig
+import org.greenstand.android.TreeTracker.viewmodels.ConfigViewModel
+import org.koin.android.ext.android.inject
+
+class ConfigFragment : Fragment() {

```

Comment:
For this config screen, I'm wondering if it would be worth making a reusable view that lets us update a value, then we specify a list of PrefKeys (with some additional data) in the ViewModel that will populate the screen. Editing a value can automatically update that PrefKey value.

This would allow us to add more configurable values by simply adding a PrefKey to the list.

Another option could be to use https://developer.android.com/reference/androidx/preference/package-summary

Reply:
I wasn't aware and familiar with this package and its uses, thank you for pointing it. I would agree it is worth to look at this in a subsequent iteration.  I prefer to have the current work go out to make use of this feature for now though.

I too thought about having a dynamic way to produce the configuration screen where specifying a PrefKey would automatically generate the UI by iterating on all the keys and values from the reads. But this would soon become complex since updating one Prefkey at a time would have to trigger a call back to effect its changed value at its point of use (instead of refreshing a cached config object).  Also,  in this specific use case, since location updates causes the config values to be used every second, it is better not to read from preferences directly but rather from a cached value. Given all this, I went the current implementation.  

I am sure there is room for improvement as we add more config objects under here and we could revisit this and examine preference screens.

Reply:
Android preferences creates its own internal cache, so it's safe to read directly. But yes, let's make any changes in another iteration.